test_conformance/api/test_queries.cpp

```diff
@@ -476,30 +478,180 @@ REGISTER_TEST(get_command_queue_info_compatibility)
+    {
         // Data must be NULL
-        if (props != 0) {
+        if (props != 0)
+        {
             log_error("ERROR: Returned properties is no NULL.\n");
             return -1;
         }
         // Valid data and size
         return 0;
     }
     // Size was not 0 or 1
-    log_error( "ERROR: Returned size of context props is not valid! (expected 0 or %d, got %d)\n",
-              (int)sizeof(cl_context_properties), (int)size );
+    log_error("ERROR: Returned size of context props is not valid! (expected 0 "
+              "or %d, got %d)\n",
+              (int)sizeof(cl_context_properties), (int)size);
     return -1;
 }
 
+REGISTER_TEST(get_context_info_mult_devices)
+{
+    cl_int err = CL_SUCCESS;
+    size_t size = 0;
+
+    // query multi-device context and perform objects comparability test
+    err = clGetDeviceInfo(device, CL_DEVICE_PARTITION_PROPERTIES, 0, nullptr,
+                          &size);
+    test_error_ret(err, "clGetDeviceInfo failed", TEST_FAIL);
+
+    test_error_ret(size == 0, "Can't partition device, test not supported\n",
+                   TEST_SKIPPED_ITSELF);
+
+    std::vector<cl_device_partition_property> supported_props(
+        size / sizeof(cl_device_partition_property), 0);
+    err = clGetDeviceInfo(device, CL_DEVICE_PARTITION_PROPERTIES,
+                          supported_props.size()
+                              * sizeof(cl_device_partition_property),
+                          supported_props.data(), &size);
+    test_error_ret(err, "clGetDeviceInfo failed", TEST_FAIL);
+
+    test_error_ret(supported_props.empty() || supported_props.front() == 0,
+                   "Can't partition device, test not supported\n",
+                   TEST_SKIPPED_ITSELF);
+
+    cl_uint maxComputeUnits = 0;
+    err = clGetDeviceInfo(device, CL_DEVICE_MAX_COMPUTE_UNITS,
+                          sizeof(maxComputeUnits), &maxComputeUnits, NULL);
+    test_error_ret(err, "Unable to get maximal number of compute units",
+                   TEST_FAIL);
+
+    std::vector<std::array<cl_device_partition_property, 5>> partition_props = {
+        { CL_DEVICE_PARTITION_EQUALLY, (cl_int)maxComputeUnits / 2, 0, 0, 0 },
+        { CL_DEVICE_PARTITION_BY_COUNTS, 1, (cl_int)maxComputeUnits - 1,
+          CL_DEVICE_PARTITION_BY_COUNTS_LIST_END, 0 },
+        { CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+          CL_DEVICE_AFFINITY_DOMAIN_NUMA, 0, 0, 0 },
+        { CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+          CL_DEVICE_AFFINITY_DOMAIN_L4_CACHE, 0, 0, 0 },
+        { CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+          CL_DEVICE_AFFINITY_DOMAIN_L3_CACHE, 0, 0, 0 },
+        { CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+          CL_DEVICE_AFFINITY_DOMAIN_L2_CACHE, 0, 0, 0 },
+        { CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+          CL_DEVICE_AFFINITY_DOMAIN_L1_CACHE, 0, 0, 0 },
+        { CL_DEVICE_PARTITION_BY_AFFINITY_DOMAIN,
+          CL_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE, 0, 0, 0 }
```

Comment:
We should be fine with just the last entry here, for `NEXT_PARTITIONABLE`.  This will partition by whatever the "next" affinity domain is, meaning we don't need to check for each of the other affinity domains specifically.

Reply:
Done